### PR TITLE
xyce: Disable CMake test for OneAPI

### PR DESCRIPTION
**Pull request by kuberry**
- Removes a call to `check_include_file_cxx` with the OneAPI compiler
- Adds a typename to a line that previously did not compile with OneAPI

@eugeneswalker

### CODE REVIEW

**Comment by eugeneswalker** [comment]
This PR fixes https://github.com/spack/spack/issues/38569 for me!
```
==> Installing xyce-7.6.0-sttmp2khzbnjtzv3p3rcb7qvd4cskwa4
==> No binary for xyce-7.6.0-sttmp2khzbnjtzv3p3rcb7qvd4cskwa4 found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/fc/fc25557e2edc82adbe0436a15fca2929a2f9ab08ddf91f1a47aab5e8b27ec88c.tar.gz
==> Fetching https://mirror.spack.io/_source-cache/archive/07/077f91d2ff0649b3f7e83c224f71a030521c6fb5a84b29acd772d5657cdb6c23
==> Applied patch /spack/var/spack/repos/builtin/packages/xyce/450-mpich-xyce.patch
==> Applied patch https://github.com/xyce/xyce/commit/fdf457fce1b1511b8a29d134d38e515fb7149246.patch?full_index=1
==> Applied patch /spack/var/spack/repos/builtin/packages/xyce/454-oneapi-xyce.patch
==> xyce: Executing phase: 'cmake'
==> xyce: Executing phase: 'build'
==> xyce: Executing phase: 'install'
==> xyce: Successfully installed xyce-7.6.0-sttmp2khzbnjtzv3p3rcb7qvd4cskwa4
  Stage: 2.20s.  Cmake: 11.29s.  Build: 1m 23.41s.  Install: 8.28s.  Post-install: 0.13s.  Total: 1m 45.56s
```

Thanks!

**Review by eugeneswalker — APPROVED** [review]

**Comment by eugeneswalker** [comment]
Should we wait for review from TBird or merge now?

**Review by TBird2001 — APPROVED** [review]
I think this is fine for now, although I find it an odd failure. But, then again, the test is not as clean as it should. The Xyce CMake build system is going to start being refined again, soon, so we will look at that in more detail then.

**Pull request merged**
